### PR TITLE
feat: Update the implementation of hash_map_open_addressing.js and .ts

### PR DESCRIPTION
**Pull request by krahets**
If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference code.
- [x] I've checked the code (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

TypeScript code is not tested.

### CODE REVIEW

File: codes/javascript/chapter_hashing/hash_map_open_addressing.js [hunk 1]
```diff
@@ -14,149 +14,157 @@ class Pair {
...
+        this.loadThres = 2.0 / 3.0; // 触发扩容的负载因子阈值
+        this.extendRatio = 2; // 扩容倍数
+        this.buckets = Array(this.capacity).fill(null); // 桶数组
+        this.TOMBSTONE = new Pair(-1, '-1'); // 删除标记
     }
 
     /* 哈希函数 */
-    #hashFunc(key) {
-        return key % this.#capacity;
+    hashFunc(key) {
```
Comment (justin-tse) [inline codes/javascript/chapter_hashing/hash_map_open_addressing.js:28]: This is a private method, so we need to add # before the method, the other private methods need to add it too, thanks!

File: codes/javascript/chapter_hashing/hash_map_open_addressing.js [hunk 2]
```diff
@@ -14,149 +14,157 @@ class Pair {
...
-        this.#loadThres = 2.0 / 3.0;
-        this.#extendRatio = 2;
-        this.#buckets = new Array(this.#capacity).fill(null);
-        this.#removed = new Pair(-1, '-1');
+        this.size = 0; // 键值对数量
+        this.capacity = 4; // 哈希表容量
+        this.loadThres = 2.0 / 3.0; // 触发扩容的负载因子阈值
+        this.extendRatio = 2; // 扩容倍数
+        this.buckets = Array(this.capacity).fill(null); // 桶数组
+        this.TOMBSTONE = new Pair(-1, '-1'); // 删除标记
```
Comment (justin-tse) [inline codes/javascript/chapter_hashing/hash_map_open_addressing.js:24]: These are all private fields, so we need to add # before them. When we use private fields, we need to define them before using them in the constructor, please reset them, thanks!

**Review by justin-tse — COMMENTED** [review]
@krahets Thank you for your clean code! Only a little problem need to update.

**Comment by krahets** [comment]
@justin-tse The comments have been addressed. Thx!

**Pull request merged**